domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/filesystem/symbolicLinkResolving.kt

```diff
@@ -0,0 +1,101 @@
+package ch.tutteli.atrium.domain.robstoll.lib.creating.filesystem
+
+import ch.tutteli.atrium.assertions.Assertion
+import ch.tutteli.atrium.assertions.AssertionGroup
+import ch.tutteli.atrium.domain.builders.ExpectImpl
+import ch.tutteli.atrium.reporting.RawString
+import ch.tutteli.atrium.translations.DescriptionPathAssertion.FAILURE_DUE_TO_LINK_LOOP
+import ch.tutteli.atrium.translations.DescriptionPathAssertion.HINT_FOLLOWED_SYMBOLIC_LINK
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.*
+
+inline fun explainForResolvedLink(path: Path, block: (realPath: Path) -> Assertion): Assertion {
+    val hintList = LinkedList<Assertion>()
+    val realPath = addAllLevelResolvedSymlinkHints(
+        path.toAbsolutePath(),
+        hintList
+    )
+    val blockResult = block(realPath)
+    return if (hintList.isNotEmpty()) {
+        when (blockResult) {
+            is AssertionGroup -> hintList.addAll(blockResult.assertions)
```

Comment:
That seems smelly to me. An AssertionGroup has more than just the assertions. Why not just adding the group?

Reply:
I am trying to solve the following problem here:

The `block` will sometimes create an `ExplanatoryAssertionGroup`. If I include it directly, it will be indented, which I do not want. This is why I unwrap the `AssertionGroup` here.

You you have a better solution for this? I understand that the type check for `AssertionGroup` is not a good solution.